### ISSUE TITLE
User can't easily focus on input field on `MenuProvider` with input - keyboard accessibility

### ISSUE THREAD

**Issue by spli02** [fortanix/baklava#380]
Users cannot easily focus on the input field inside of `MenuProvider` with input elements.

 Currently, when the dropdown opens, the first option receives focus, even though the input field appears before it.
 To focus on the input field, users must press the Tab key instead of using the arrow keys.

Ideally, the input field should receive focus when the dropdown opens, if it appears before the options.
Users should also be able to navigate to the input field using arrow keys, not the Tab key.

https://github.com/user-attachments/assets/d5c6eba7-ad41-4c1c-8d46-745a5fc52516
